### DISCUSSION TITLE
Applying raw + mask data as a training dataset

### DISCUSSION THREAD

**Discussion by joowson** [jeanollion/bacmman#17] [General]
Hello, 

At first, I'd like to say thank you for sharing this impressive method along with the thorough documentation. I recently came across your publication in PRXLife, and now I am exploring how to adopt DiSTNet2D as a reference model. In this process, I am interested in training neural net with my training dataset, but I haven't been able to find how to do this in the documentation. I wonder how to do this if it is possible.

Could you please advise on how to train the model with my custom data, which consists of raw 2D images of swimming bacteria and their corresponding masks array? (see image below) The goal is to avoid the manual labeling process by using an existing answer dataset. I would appreciate any insights on how to proceed, whether via the BACMMAN GUI or command line.

This would greatly help my work, not only in saving time but also in light of the fact that the one of main focuses of my current research is to automatically generate a training dataset for 2D bacterial images.

Sincerely,

Joowang Son, PhD. Student
Ulsan National Institute of Science and Technology, Korea

![raw+mask 2](https://github.com/user-attachments/assets/ad845438-a412-4b1b-9f92-27e10877e38d)
